cli/tests/test_checkout.rs

```diff
@@ -158,6 +158,23 @@ fn test_checkout_conflicting_change_ids() {
     "###);
 }
 
+#[test]
+fn test_checkout_non_empty_working_copy() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
```

Comment:
Maybe you mean `--git-repo=` to set up a colocated repo? Non-colocated repo won't have `HEAD@git`. 

You can copy setup and assertion codes from test_git_colocated.rs

Reply:
I moved the test to `test_git_colocated.rs` as that seemed to make more sense, but I'm not sure what assertions we want here.

Reply:
I don't know what's supposed to be covered by this test, but maybe you can check if the `HEAD` still point to the original branch?

`git_repo.find_reference("HEAD").unwrap().symbolic_target()`

Reply:
When I try to create a branch, I get
```raw
    Failed to export some branches:
      master
```
I think this is causing a panic when I try to unwrap the target. Is there anything I'm missing?

Reply:
Which commit does the branch point to? Common failure is that the branch is set to the root commit, but in that case, you'll probably get more friendly error message.

Anyway, you'll probably need to use `git_repo` (instead of `["jj", "branch", ...]`) to make the HEAD undetached.

Reply:
I simply copied what's on `test_git_colocated` to create the first commit via git so we get everything right. Not sure if there's a simpler way to do that.